src/renderer/store/modules/transaction.js

```diff
@@ -142,6 +142,58 @@ export default {
       return data
     },
 
+    processVotes ({ dispatch, rootGetters }, transactions = []) {
+      const unconfirmedVotes = rootGetters['session/unconfirmedVotes']
+      const profile = rootGetters['session/profile']
+
+      if (!unconfirmedVotes || !unconfirmedVotes.length || !profile) {
+        return
+      }
+
+      const votes = transactions.filter(tx => tx.type === config.TRANSACTION_TYPES.VOTE)
+      if (!votes.length) {
+        return
+      }
+
+      const ids = votes.map(vote => vote.id)
+      const pendingVotes = unconfirmedVotes.filter(vote => {
+        return !ids.includes(vote.id)
+      })
+
+      dispatch('session/setUnconfirmedVotes', pendingVotes, { root: true })
+
+      dispatch('profile/update', {
+        ...profile,
+        unconfirmedVotes: pendingVotes
+      }, { root: true })
+    },
+
+    clearUnconfirmed ({ dispatch, rootGetters }) {
+      const unconfirmedVotes = rootGetters['session/unconfirmedVotes']
+      const profile = rootGetters['session/profile']
+
+      if (!unconfirmedVotes || !unconfirmedVotes.length || !profile) {
+        return
+      }
+
+      const pendingVotes = unconfirmedVotes.filter(vote => {
+        if (!vote.timestamp) {
+          return true
+        }
+        const threshold = dayjs(vote.timestamp).add(6, 'hour')
+        if (dayjs(vote.timestamp).isBefore(threshold)) {
```

Comment:
https://github.com/ArkEcosystem/desktop-wallet/blob/829d14a59313a31dd4b00b4b90f77bb70b741a02/src/renderer/store/modules/transaction.js#L184-L187 could be reduced to

```
        return !dayjs(vote.timestamp).isBefore(threshold)
```